### ISSUE TITLE
[Bug]:  Concurrently search gets exception channel is not available in any replica

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#24388]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20230519-4500d632
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.4.0.dev27
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy milvus with config
```
{'queryNode': {'resources': {'limits': {'cpu': '2.0',
                                                              'memory': '12Gi'},
                                                   'requests': {'cpu': '2.0',
                                                                'memory': '7Gi'}}},
                       'indexNode': {'resources': {'limits': {'cpu': '2.0',
                                                              'memory': '12Gi'},
                                                   'requests': {'cpu': '2.0',
                                                                'memory': '7Gi'}}},
                       'dataNode': {'resources': {'limits': {'cpu': '2.0',
                                                             'memory': '12Gi'},
                                                  'requests': {'cpu': '2.0',
                                                               'memory': '7Gi'}}},
                       'cluster': {'enabled': True},
                       'pulsar': {'bookkeeper': {'volumes': {'journal': {'storageClassName': 'local-path'},
                                                             'ledgers': {'storageClassName': 'local-path'}}},
                                  'zookeeper': {'volumes': {'data': {'storageClassName': 'local-path'}}}},
                       'kafka': {'persistence': {'storageClass': 'local-path'}},
                       'minio': {'persistence': {'storageClass': 'local-path'},
                                 'metrics': {'serviceMonitor': {'enabled': False}}},
                       'etcd': {'global': {'storageClass': 'local-path'},
                                'metrics': {'enabled': True,
                                            'podMonitor': {'enabled': True}}},
                       'metrics': {'serviceMonitor': {'enabled': True}},
                       'log': {'level': 'debug'},
                       'image': {'all': {'repository': 'harbor.milvus.io/milvus/milvus',
                                         'tag': 'master-20230519-4500d632'}}}
```

2. client case params:
```
{'dataset_params': {'metric_type': 'L2',
                                                    'dim': 128,
                                                    'dataset_name': 'sift',
                                                    'dataset_size': 1000000,
                                                    'ni_per': 50000},
                                 'collection_params': {'other_fields': [],
                                                       'shards_num': 2},
                                 'load_params': {},
                                 'query_params': {},
                                 'search_params': {},
                                 'resource_groups_params': {'reset': False},
                                 'database_user_params': {'reset_rbac': False,
                                                          'reset_db': False},
                                 'index_params': {'index_type': 'IVF_SQ8',
                                                  'index_param': {'nlist': 1024}},
                                 'concurrent_params': {'concurrent_number': 20,
                                                       'during_time': '1h',
                                                       'interval': 20,
                                                       'spawn_rate': None},
                                 'concurrent_tasks': [{'type': 'search',
                                                       'weight': 7,
                                                       'params': {'nq': 1,
                                                                  'top_k': 10,
                                                                  'search_param': {'nprobe': 16},
                                                                  'expr': None,
                                                                  'guarantee_timestamp': None,
                                                                  'output_fields': None,
                                                                  'ignore_growing': False,
                                                                  'timeout': 300,
                                                                  'random_data': True}},
                                                      {'type': 'scene_insert_delete_flush',
                                                       'weight': 1,
                                                       'params': {'insert_length': 10,
                                                                  'delete_length': 1,
                                                                  'random_id': True,
                                                                  'random_vector': True,
                                                                  'varchar_filled': True}}]}
```

3. Test steps
- create collection with 2 shards
- create index `params:{'index_type': 'IVF_SQ8', 'metric_type': 'L2', 'params': {'nlist': 1024}}`
- insert 1m 128d data with ni_per 5w
- flush
- create index again
- load
- concurrent search and scene_insert_delete_flush, after a while search return exception:
```
[2023-05-19 08:21:12,160 - ERROR - fouram]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)>, <Time:{'RPC start': '2023-05-19 08:20:45.280798', 'RPC error': '2023-05-19 08:21:12.160254'}> (decorators.py:108)
[2023-05-19 08:21:12,163 - ERROR - fouram]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)> (api_request.py:49)
[2023-05-19 08:21:12,163 - ERROR - fouram]: [CheckFunc] search request check failed, response:<MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)> (func_check.py:46)
[2023-05-19 08:21:12,166 - ERROR - fouram]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)>, <Time:{'RPC start': '2023-05-19 08:20:45.138296', 'RPC error': '2023-05-19 08:21:12.166156'}> (decorators.py:108)
[2023-05-19 08:21:12,166 - ERROR - fouram]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)> (api_request.py:49)
[2023-05-19 08:21:12,166 - ERROR - fouram]: [CheckFunc] search request check failed, response:<MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)> (func_check.py:46)
[2023-05-19 08:21:12,167 - ERROR - fouram]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)>, <Time:{'RPC start': '2023-05-19 08:20:45.268148', 'RPC error': '2023-05-19 08:21:12.167195'}> (decorators.py:108)
[2023-05-19 08:21:12,167 - ERROR - fouram]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)> (api_request.py:49)
[2023-05-19 08:21:12,167 - ERROR - fouram]: [CheckFunc] search request check failed, response:<MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)> (func_check.py:46)
[2023-05-19 08:21:12,168 - ERROR - fouram]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_441577280729776304v1 is not available in any replica, err=<nil>: attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_0_441577280729776304v0 is not available in any replica, err=<nil>: context done during sleep after run#6: context deadline exceeded)>, <Time:{'RPC start': '2023-05-19 08:20:45.162340', 'RPC error': '2023-05-19 08:21:12.168237'}> (decorators.py:108)
```

### Steps To Reproduce

```markdown
argo workflow name in devops cluster: `fouramf-issue4`
```

### Milvus Log

pods in devops cluster chaos-testing namespace:
```
fouramf-issue4-68-1234-etcd-0                                     1/1     Running            1 (5m10s ago)      6m59s   10.102.6.238    devops-node10   <none>           <none>
fouramf-issue4-68-1234-etcd-1                                     1/1     Running            0                  6m59s   10.102.9.157    devops-node13   <none>           <none>
fouramf-issue4-68-1234-etcd-2                                     1/1     Running            0                  6m58s   10.102.7.55     devops-node11   <none>           <none>
fouramf-issue4-68-1234-milvus-datacoord-59d5999566-676g7          1/1     Running            1 (2m58s ago)      6m59s   10.102.9.129    devops-node13   <none>           <none>
fouramf-issue4-68-1234-milvus-datanode-5d648f6d88-s6tfg           1/1     Running            1 (2m41s ago)      6m59s   10.102.9.151    devops-node13   <none>           <none>
fouramf-issue4-68-1234-milvus-indexcoord-8df69cf69-qt2d8          1/1     Running            0                  6m59s   10.102.9.126    devops-node13   <none>           <none>
fouramf-issue4-68-1234-milvus-indexnode-d87b79cd7-422pv           1/1     Running            0                  6m59s   10.102.9.132    devops-node13   <none>           <none>
fouramf-issue4-68-1234-milvus-proxy-7d75dccf9-l2vxx               1/1     Running            1 (2m59s ago)      6m59s   10.102.9.125    devops-node13   <none>           <none>
fouramf-issue4-68-1234-milvus-querycoord-6f96d4bb6d-x278s         1/1     Running            1 (2m58s ago)      6m59s   10.102.6.209    devops-node10   <none>           <none>
fouramf-issue4-68-1234-milvus-querynode-7d9f6cc9f7-5clrr          1/1     Running            0                  6m59s   10.102.6.211    devops-node10   <none>           <none>
fouramf-issue4-68-1234-milvus-rootcoord-58c65fcb64-qw9vx          1/1     Running            1 (2m58s ago)      6m59s   10.102.9.131    devops-node13   <none>           <none>
fouramf-issue4-68-1234-minio-0                                    1/1     Running            0                  6m59s   10.102.6.228    devops-node10   <none>           <none>
fouramf-issue4-68-1234-minio-1                                    1/1     Running            0                  6m59s   10.102.9.154    devops-node13   <none>           <none>
fouramf-issue4-68-1234-minio-2                                    1/1     Running            0                  6m59s   10.102.7.51     devops-node11   <none>           <none>
fouramf-issue4-68-1234-minio-3                                    1/1     Running            0                  6m58s   10.102.5.59     devops-node21   <none>           <none>
fouramf-issue4-68-1234-pulsar-bookie-0                            1/1     Running            0                  6m59s   10.102.6.226    devops-node10   <none>           <none>
fouramf-issue4-68-1234-pulsar-bookie-1                            1/1     Running            0                  6m59s   10.102.9.159    devops-node13   <none>           <none>
fouramf-issue4-68-1234-pulsar-bookie-2                            1/1     Running            0                  6m58s   10.102.7.56     devops-node11   <none>           <none>
fouramf-issue4-68-1234-pulsar-bookie-init-97jfc                   0/1     Completed          0                  6m59s   10.102.6.212    devops-node10   <none>           <none>
fouramf-issue4-68-1234-pulsar-broker-0                            1/1     Running            0                  6m59s   10.102.9.136    devops-node13   <none>           <none>
fouramf-issue4-68-1234-pulsar-proxy-0                             1/1     Running            0                  6m59s   10.102.9.133    devops-node13   <none>           <none>
fouramf-issue4-68-1234-pulsar-pulsar-init-b4tcb                   0/1     Completed          0                  6m59s   10.102.6.208    devops-node10   <none>           <none>
fouramf-issue4-68-1234-pulsar-recovery-0                          1/1     Running            0                  6m59s   10.102.9.134    devops-node13   <none>           <none>
fouramf-issue4-68-1234-pulsar-zookeeper-0                         1/1     Running            0                  6m59s   10.102.9.153    devops-node13   <none>           <none>
fouramf-issue4-68-1234-pulsar-zookeeper-1                         1/1     Running            0                  5m26s   10.102.6.248    devops-node10   <none>           <none>
fouramf-issue4-68-1234-pulsar-zookeeper-2                         1/1     Running            0                  4m47s   10.102.7.58     devops-node11   <none>           <none> (cli_client.py:131)
```

**Comment by weiliu1031** [comment]
same issue as #23936
already fixed by #24300

**Comment by weiliu1031** [comment]
@ThreadDao  please verify this

**Comment by ThreadDao** [comment]
Fixed `master-20230529-d8ad09b7`
argo workflow in devops: `fouramf-issue5`
